### PR TITLE
Feature: Add Invalid Archive Name & Invalid Shortcut Location Warning

### LINKED ISSUE #12044: Feature: Add invalid name and path warnings to modals
### What feature or improvement do you think would benefit Files?

At the moment, when you try to create a shortcut, if you enter a non existing path, `Create` button will be disabled but no reason is shown.
Thre's a similar issue when creating archives: you can enter whatever name you want (even using forbidden chars), and the `Create` button will always be enabled. If the name contains forbidden chars, the creation of the archive will fail and no reason will be given (only `Couldn't create the archive`)
Finally, there's a warning tip when we you try to create a file with an invalid name. This is a simple text block and looks different from the one used in tag creation. I think we should always use the same style to be consistent

### Requirements

- Add a warning tip when the shortcut path doesn't exist
- Add a warning tip when the archive name isn't valid
- Change the style of the warning tip of the `Rename` dynamic dialog 

### Files Version

2.4.61.0

### Windows Version

Windows 11 Pro 22H2 10.0.22621.1485

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12044 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app and try to create an archive
   2. Clear archive name

**Screenshots (optional)**

### CODE REVIEW

**Comment by yaira2** [comment]
Can you link an issue for this?

**Comment by ferrariofilippo** [comment]
I'm having some troubles with `rename dialog` because of an exception, I'll try to find a workaround

File: src/Files.App/Dialogs/CreateArchiveDialog.xaml [hunk 1]
```diff
@@ -29,11 +35,16 @@
 			BorderThickness="1"
 			ColumnSpacing="8"
 			CornerRadius="4">
+			<Grid.RowDefinitions>
+				<RowDefinition Height="Auto" />
+				<RowDefinition Height="Auto" />
+			</Grid.RowDefinitions>
```
Comment (yaira2) [inline src/Files.App/Dialogs/CreateArchiveDialog.xaml:41]: Is this still needed?

Comment (ferrariofilippo) [reply src/Files.App/Dialogs/CreateArchiveDialog.xaml:41]: No, I'll remove it

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**